portable/CMakeLists.txt

```diff
@@ -0,0 +1,276 @@
+    $<$<STREQUAL:${FREERTOS_PORT},OWATCOM_16BIT_DOS_FLSH186>:oWatcom/16BitDOS/common/portcomn.c;oWatcom/16BitDOS/Flsh186/port.c>
+    $<$<STREQUAL:${FREERTOS_PORT},OWATCOM_16BIT_DOS_PC>:oWatcom/16BitDOS/common/portcomn.c;oWatcom/16BitDOS/PC/port.c>
+    $<$<STREQUAL:${FREERTOS_PORT},PARADIGM_TERN_EE_LARGE>:Paradigm/Tern_EE/large_untested/port.c>
+    $<$<STREQUAL:${FREERTOS_PORT},PARADIGM_TERN_EE_SMALL>:Paradigm/Tern_EE/small/port.c>
+    $<$<STREQUAL:${FREERTOS_PORT},RENESAS_RX100>:Renesas/RX100/port.c;Renesas/RX100/port_asm.src>
+    $<$<STREQUAL:${FREERTOS_PORT},RENESAS_RX200>:Renesas/RX200/port.c;Renesas/RX200/port_asm.src>
+    $<$<STREQUAL:${FREERTOS_PORT},RENESAS_RX600>:Renesas/RX600/port.c;Renesas/RX600/port_asm.src>
+    $<$<STREQUAL:${FREERTOS_PORT},RENESAS_RX600_V2>:Renesas/RX600v2/port.c;Renesas/RX600v2/port_asm.src>
+    $<$<STREQUAL:${FREERTOS_PORT},RENESAS_RX700_V3_DPFPU>:Renesas/RX700v3_DPFPU/port.c;Renesas/RX700v3_DPFPU/port_asm.src>
+    $<$<STREQUAL:${FREERTOS_PORT},RENESAS_SH2A_FPU>:Renesas/SH2A_FPU/port.c;Renesas/SH2A_FPU/portasm.src>
+    $<$<STREQUAL:${FREERTOS_PORT},ROWLEY_MSP430F449>:Rowley/MSP430F449/port.c;Rowley/MSP430F449/portext.asm>
+    $<$<STREQUAL:${FREERTOS_PORT},RVDS_ARM_CA9>:RVDS/ARM_CA9/port.c;RVDS/ARM_CA9/portASM.s>
+    $<$<STREQUAL:${FREERTOS_PORT},RVDS_ARM_CM0>:RVDS/ARM_CM0/port.c>
+    $<$<STREQUAL:${FREERTOS_PORT},RVDS_ARM_CM3>:RVDS/ARM_CM3/port.c>
+    $<$<STREQUAL:${FREERTOS_PORT},RVDS_ARM_CM4_MPU>:RVDS/ARM_CM4_MPU/port.c>
+    $<$<STREQUAL:${FREERTOS_PORT},RVDS_ARM_CM4F>:RVDS/ARM_CM4F/port.c>
+    $<$<STREQUAL:${FREERTOS_PORT},RVDS_ARM_CM7>:RVDS/ARM_CM7/r0p1/port.c>
+    $<$<STREQUAL:${FREERTOS_PORT},RVDS_ARM7_LPC21XX>:RVDS/ARM7_LPC21xx/port.c;RVDS/ARM7_LPC21xx/portASM.s>
+    $<$<STREQUAL:${FREERTOS_PORT},SDCC_CYGNAL>:SDCC/Cygnal/port.c>
+    $<$<STREQUAL:${FREERTOS_PORT},SOFTUNE_MB91460>:Softune/MB91460/__STD_LIB_sbrk.c;Softune/MB91460/port.c>
+    $<$<STREQUAL:${FREERTOS_PORT},SOFTUNE_MB96340>:Softune/MB96340/__STD_LIB_sbrk.c;Softune/MB96340/port.c>
+    $<$<STREQUAL:${FREERTOS_PORT},TASKING_ARM_CM4F>:Tasking/ARM_CM4F/port.c;Tasking/ARM_CM4F/port_asm.asm>
+    $<$<STREQUAL:${FREERTOS_PORT},CDK_THEAD_CK802>:ThirdParty/CDK/T-HEAD_CK802/port.c;ThirdParty/CDK/T-HEAD_CK802/portasm.S>
+    $<$<STREQUAL:${FREERTOS_PORT},XCC_XTENSA>:ThirdParty/XCC/Xtensa/port.c;ThirdParty/XCC/Xtensa/portasm.S;ThirdParty/XCC/Xtensa/portclib.c;ThirdParty/XCC/Xtensa/xtensa_context.S;ThirdParty/XCC/Xtensa/xtensa_init.c;ThirdParty/XCC/Xtensa/xtensa_intr_asm.S;ThirdParty/XCC/Xtensa/xtensa_intr.c;ThirdParty/XCC/Xtensa/xtensa_overlay_os_hook.c;ThirdParty/XCC/Xtensa/xtensa_vectors.S>
+    $<$<STREQUAL:${FREERTOS_PORT},WIZC_PIC18>:WizC/PIC18/port.c;WizC/PIC18/Drivers/Tick/isrTick.c;WizC/PIC18/Drivers/Tick/Tick.c>
+)
+
+target_include_directories(freertos_kernel_port PUBLIC
+    $<$<STREQUAL:${FREERTOS_PORT},BCC_16BIT_DOS_FLSH186>:${CMAKE_CURRENT_LIST_DIR}/BCC/16BitDOS/common;${CMAKE_CURRENT_LIST_DIR}/BCC/16BitDOS/Flsh186>
```

Comment:
Do we need to include kernel headers and `FreeRTOSConfig.h`?
I see that windows `port.c` includes `FreeRTOS.h` and `task.h`. I have not tried to build this but was just wondering.

Reply:
Yes, we need. They are included for `freertos_kernel_port` here: https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/421/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR195

All public include paths which belongs to `freertos_kernel` target initially, propagated to `freertos_kernel_port` here:
```
target_link_libraries(freertos_kernel freertos_kernel_port)
```

Reply:
Oh yes, I missed that detail. Thank you for the clarification. I'll approve the PR.